src/pe_crypto.cpp

```diff
@@ -101,85 +103,99 @@ namespace libtorrent
 		m_xor_mask = h.final();
 	}
 
-	int encryption_handler::encrypt(std::vector<boost::asio::mutable_buffer>& iovec)
+	std::tuple<int, std::vector<boost::asio::const_buffer>>
+	encryption_handler::encrypt(
+		aux::array_view<boost::asio::mutable_buffer> iovec)
```

Comment:
This is probably the largest relevant hunk of this patch. I'll leave some comments.